packages/hierarchies/api/presentation-hierarchies.api.md

```diff
@@ -26,7 +26,7 @@ interface BaseHierarchyNode {
         [key: string]: any;
     };
     filtering?: {
-        isFilterTarget?: boolean;
+        filterTarget?: FilterTarget;
```

Comment:
I feel like it's not completely correct to set grouping node's information to a "filter target" prop - filter target should always be the end of a filtering path. The grouping node is not a filter target, but instead and "auto expand target". What do thou think about keeping the `isFilterTarget` prop and additionally having the `autoExpand` one?

Reply:
Since this change was done by @jasdom, and you've said that everything looks OK, I didn't want to change it. Although, now that you've mentioned it, I agree with your suggestion. 

Come to think of it, why is `isFilterTarget` or even `filtering` property a part of `BaseHierarchyNode`, if grouping nodes can never be filter targets?

Reply:
What do you think about this?
```ts
filtering?: {
    hasFilterTargetAncestor?: boolean;
    filteredChildrenIdentifierPaths?: HierarchyFilteringPath[];
  } & (
    | { isFilterTarget: never }
    | {
        isFilterTarget: true;
        autoExpandUntil?: FilterTargetGroupingNodeInfo;
      }
  );
  ```

Reply:
> why is `isFilterTarget` or even `filtering` property a part of `BaseHierarchyNode`, if grouping nodes can never be filter targets?

Good question - probably they shouldn't be.

> What do you think about this?

The idea looks good. Having `{ isFilterTarget: never }` may be inconvenient to use, e.g. I think you won't be able to do something like `isFilterTarget: x > 10`. Maybe change it to `undefined | false`?

Reply:
Changed the API to how you've recommended